types/xhook/index.d.ts

```diff
@@ -0,0 +1,44 @@
+// Type definitions for xhook 1.5
+// Project: https://github.com/jpillora/xhook#readme
+// Definitions by: Sudarsan Balaji <https://github.com/artfuldev>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export interface Request {
+    method: string;
+    url: string;
+    body: string;
+    headers: Record<string, string>;
+    timeout: number;
+    type: string;
+    withCredentials: boolean;
+}
+
+export type ResponseType = ArrayBuffer | Document | Blob | object | string;
+
+export interface Response {
+    status: number;
+    statusText: string;
+    text: string;
+    headers: Record<string, string>;
+    xml: XMLDocument;
+    data: ResponseType;
+}
+
+export interface BeforeHandler {
+    (request: Request, callback: (response?: Response) => void): Response | void;
+}
+
+export interface AfterHandler {
+    (request: Request, response: Response, callback: () => void): void;
+}
+
+interface XHook {
+    enable: () => void;
+    disable: () => void;
+    before: (handler: BeforeHandler, index?: number) => void;
+    after: (handler: AfterHandler, index?: number) => void;
+}
+
+declare const xhook: XHook;
+
+export default xhook;
```

Comment:
looking at xhook's source, it exports a single object using commonjs, not an object with a `default` property. So you should `export = xhook` here, and move all the types into a `namespace xhook` which will merge with `const xhook`.

Reply:
Thank you so much for this!

I'm not sure I understand the fix - I've attempted a version with a namespace and a const - but keeping the type of the const local and not exported. Is there a better way to describe the same?

Reply:
I've removed the local interface. Hopefully that's better.